### ISSUE TITLE
Release a new `4337` module version with updated JS tooling.

### ISSUE THREAD

**Issue by mmv08** [safe-global/safe-modules#426]
## Context / issue

I needed to deploy & verify the 4337 contract to a new network, so I checked out to `[4337/v0.3.0](https://github.com/safe-global/safe-modules/tree/4337/v0.3.0)` tag.

And ran into an issue with installing dependencies:
```
~/projects/safe/safe-modules #4337/v0.3.0@5931a275 ❯ npm i                                                                                                                                               22s 17:22:01
npm error code 1
npm error path /Users/mmv/projects/safe/safe-modules/modules/4337
npm error command failed
npm error command sh -c npm run build
npm error > @safe-global/safe-4337@0.3.0 build
npm error > npm run build:sol && npm run build:ts
npm error
npm error
npm error > @safe-global/safe-4337@0.3.0 build:sol
npm error > rimraf build typechain-types && hardhat compile
npm error Error HH18: You installed Hardhat with a corrupted lockfile due to the NPM bug #4828.
npm error
npm error Please delete your node_modules, package-lock.json, reinstall your project, and try again.
npm error For more info go to https://hardhat.org/HH18 or run Hardhat with --show-stack-traces
npm error npm error Lifecycle script `build:sol` failed with error:
npm error npm error Error: command failed
npm error npm error   in workspace: @safe-global/safe-4337@0.3.0
npm error npm error   at location: /Users/mmv/projects/safe/safe-modules/modules/4337
npm error npm error Lifecycle script `build` failed with error:
npm error npm error Error: command failed
npm error npm error   in workspace: @safe-global/safe-4337@0.3.0
npm error npm error   at location: /Users/mmv/projects/safe/safe-modules/modules/4337

npm error A complete log of this run can be found in: /Users/mmv/.npm/_logs/2024-05-27T15_22_27_743Z-debug-0.log
```

In the end, I had to use the current main branch, which contains pnpm updates (luckily, we didn't do any code changes to the 4337 contract)

## Proposed solution

Release a new version with updated js tooling and a new tag (e.g., 4337/v0.3.0-1 per @nlordell suggestion)

**Comment by akshay-ap** [comment]
My understanding is that as the script has a bug fix, new version should `v0.3.1`. Curious to know why there is `v0.3.0-1` as new version instead? I couldn't find it as a convention in npm docs.

**Comment by mmv08** [comment]
v0.3.0-1 was released https://github.com/safe-global/safe-modules/releases/tag/4337%2Fv0.3.0-1
